pkg/controller/node/node_controller.go

```diff
@@ -987,9 +990,21 @@ func getCandidateMachines(pool *mcfgv1.MachineConfigPool, nodesInPool []*corev1.
 	return nodes[:capacity]
 }
 
-// getCurrentEtcdLeader is not yet implemented
-func (ctrl *Controller) getCurrentEtcdLeader(candidates []*corev1.Node) (*corev1.Node, error) {
-	return nil, nil
+// getCurrentEtcdLeaderName fetches the name of the current node running the machine-config-operator pod
+func (ctrl *Controller) getCurrentEtcdLeaderName() (string, error) {
```

Comment:
But if it's about the MCO pod now, why not rename the function too?  The etcd leader can be on a different node than the operator pod, unless I'm missing something...

Reply:
There is an interesting topic here though in that it might actually be advantageous to us to keep the etcd leader and the operator *and* controller pods all co-located at least in a steady state, because that node should be upgraded last.

Reply:
I also like the idea of co-locating them too, I'm not sure why they were split up but I can dig into the costs/benefits of that!

Reply:
Ah...so very briefly, etcd is a replicated key-value store that is the data storage heart of Kubernetes.  Underneath, etcd uses [raft](https://github.com/etcd-io/raft) which is a generic protocol, and has the concept of a "leader".

I think what you're crossing here with that is that our operator and controller pods use a "leader election" protocol too using the apiserver (which uses etcd, which uses raft) - but this is at an entirely different level.

Reply:
Gotcha, makes sense - I'll update the function name to be more accurate. I'll also make a follow-up spike/task to explore putting the pods and the etcd-leader on the same node. Thanks!